lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -275,14 +291,31 @@ private JWEObject signEncryptJar(
             String oauthState,
             String govukSigninJourneyId,
             String criId)
-            throws HttpResponseExceptionWithErrorBody, ParseException, JOSEException {
+            throws HttpResponseExceptionWithErrorBody, ParseException, JOSEException,
+                    UnknownEvidenceTypeException {
+
+        List<String> credentials = userIdentityService.getUserIssuedCredentials(userId);
+
         SharedClaimsResponse sharedClaimsResponse =
-                getSharedAttributesForUser(ipvSessionItem, userId, criId);
+                getSharedAttributesForUser(ipvSessionItem, criId, credentials);
         EvidenceRequest evidenceRequest = null;
+
+        List<SignedJWT> signedJWTS = gpg45ProfileEvaluator.parseCredentials(credentials);
+        Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(signedJWTS);
+        List<RequiredGpg45ScoresDto> requiredGpg45Scores =
+                ACCEPTED_PROFILES.stream()
+                        .map(
+                                profile ->
+                                        new RequiredGpg45ScoresDto(
+                                                profile,
+                                                Gpg45ScoresDto.fromGpg45Scores(
+                                                        gpg45Scores.calculateRequiredScores(
+                                                                profile))))
+                        .collect(Collectors.toList());
+
         if (criId.equals(F2F_CRI)) {
             int strengthScore =
-                    gpg45ProfileEvaluator.calculateF2FRequiredStrengthScore(
-                            ipvSessionItem.getRequiredGpg45Scores());
```

Comment:
Does this mean we can remove the calculation from the `EvaluateGPG45ScoresHandler` (lines 256-267), as well as the field on the `IpvSessionItem`?

I think this was the only usage, unless you know of others?

Reply:
done. Can also remove column from ipv session table as seems not used now.

Reply:
Yep - do you think we need to wait for any active sessions to expire, or can we just do it?